### PR TITLE
[Follow-up for #4180] Avoid using mxnet 1.8.0 & Fix no-space-left error in `test-anaconda-compatibility.sh`

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

- Depends on #4180
- Avoid using `mxnet 1.8.0`
- Fix a no-space-left error in `test-anaconda-compatibility.sh`

## How is this patch tested?

Existing unit tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: dev/extra-ml-requirements.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ h5py<3.0.0
 # Required by mlflow.sklearn
 scikit-learn
 # Required by mlflow.gluon
-mxnet
+mxnet!=1.8.0
```
Comment (harupy) [inline dev/extra-ml-requirements.txt:14]: Fix for this error:

https://github.com/mlflow/mlflow/runs/2163733962#step:5:3141

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -176,6 +176,10 @@ jobs:
         export PATH="$CONDA_DIR/bin:$PATH"
         source activate test-environment
         ./dev/run-large-python-tests.sh
+    - name: Clean-up test-environment
+      run: |
+        conda remove --all --yes --name test-environment
+        conda clean --all --yes
```
Comment (harupy) [inline .github/workflows/master.yml:182]: Fix for this error in `test-anaconda-compatibility.sh`:

```
docker: failed to register layer: Error processing tar file(exit status 1): write /opt/conda/pkgs/mkl-2019.3-199.tar.bz2: no space left on device.
```

https://github.com/mlflow/mlflow/runs/2163733721#step:6:132

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -236,6 +241,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Increase available disk space
+        run: |
+          # Increase available disk space by removing unnecessary tool chains:
+          # https://github.com/actions/virtual-environments/issues/709#issuecomment-612569242
+          rm -rf "$AGENT_TOOLSDIRECTORY"
```
Comment (harupy) [inline .github/workflows/master.yml:248]: Fix for https://github.com/mlflow/mlflow/runs/2164097339#step:5:2188

Comment (harupy) [reply .github/workflows/master.yml:248]: I'm still investigating the root cause of this issue.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**